Conjur/CONTRIBUTING.md

```diff
@@ -118,7 +151,233 @@ important to follow when developing.
+- Most documentation changes
+- Addition and removal of the same feature within the same release
+
+When writing changelog entries also try to use as concise as you can to describe the change
+but don't needlessly avoid mutli-line entries.
+
+**Important**: If a changelog entry is tied to an issue, you **_must_** include a link to that
+issue as a trailing fully-expanded link so that cross-referencing can be easily done. The
+line below shows an example of an issue-linked changelog entry:
+```markdown
+- Fix support for using deployment as K8s authentication resource type for Kubernetes >= 1.16
+  ([conjur/issues#1440](https://github.com/cyberark/conjur/issues/1440))
+```
+
+Some more information about general changelog maintenance can also be found [here](https://depfu.com/blog/what-makes-a-good-changelog)
+and [here](https://docs.gitlab.com/ee/development/changelog.html).
+
+To put this all together, below are a few examples of good and bad changelog entries:
+
+**Bad entries**:
+```markdown
+...
+# Exclude documentation
+- Made links in docs use markdown format
+
+# Irrelevant to end-user
+- Added editor tempfiles to `.gitignore`
+
+# Lacking granularity and no issue link
+- Fixed issue with dependencies of the project
+
+# Irrelevant to end-user
+- Added more tests around API connectivity ([#123](https://github.com/org/repo/issues/123))
+
+# Bad issue links
+- Added FIPS compatibility to the agent (#123)
+- Added FIPS compatibility to the agent (org/repo#123)
+- Added FIPS compatibility to the agent ([#123](org/repo#123))
+- Added FIPS compatibility to the agent ([org/repo#123](#123))
+...
+```
+
+**Good entries**:
+```markdown
+...
+- Added FIPS compatibility to the agent ([org/repo#123](https://github.com/org/repo/issues/123))
+- Removed `-n` flag from the accepted CLI flags ([org/repo#234](https://github.com/org/repo/issues/234))
+- Improved performance of main API endpoints with change to
+  a new router `foo` package dependency
+- Fixed CVE-2020-12345 by upgrading `testing` dependency to v3 ([org/repo#345](https://github.com/org/repo/issues/345))
+- Fixed overrides handling of `Client` account param [org/repo#456](https://github.com/org/repo/issues/456)
+...
+```
+
+## Tagging
+
+All projects cyclically get a point where they are ready to package a group of changes as a
+check-pointed release which in most cases is started with creating versions through `git tag`.
+Git tags create special branches that allow for a reproducible and easy-to-understand
+delivery of code that are generally 1:1 mapped with significant feature changes.
+
+In most cases for our projects we use [annotated git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_annotated_tags)
+with the name matching the `v<semver>` tri-dotted pattern (e.g. `v1.2.3`). By doing this,
+we make finding relevant code and automating releases easy and simple. Where possible, we
+also use GPG signing of those tags though this process has not been yet fully designed.
```

Comment:
> though this process has not been yet fully designed
in what way? I think this comment could be clearer, or maybe even just drop this and leave as "Where possible, we do this"